src/main/java/in/org/projecteka/hiu/dataprocessor/HealthDataProcessor.java

```diff
@@ -165,23 +166,40 @@ private void notifyHealthInfoStatus(DataContext context,
                                         SessionStatus sessionStatus) {
         String consentId = dataFlowRepository.getConsentId(context.getTransactionId()).block();
         String hipId = consentRepository.getHipId(consentId).block();
-        HealthInfoNotificationRequest healthInfoNotificationRequest = HealthInfoNotificationRequest.builder()
+        HealthInfoNotificationRequest healthInfoNotificationRequest =
+                getHealthInfoNotificationRequest(context, statusResponses, sessionStatus, consentId, hipId);
+        String token = centralRegistry.token().block();
```

Comment:
Is blocking fine here?

Reply:
the method notifyHealthInfoStatus is not in reactive context. So, we didn't find any other way than blocking